sqlitecluster/SQLiteClusterMessenger.cpp

```diff
@@ -222,8 +222,20 @@ bool SQLiteClusterMessenger::runOnLeader(BedrockCommand& command) {
     // Finish our escalation timing.
     command.escalationTimeUS = STimeNow() - command.escalationTimeUS;
 
-    // Since everything went fine with this command, we can save its socket.
-    {
+    // See if either the client or the leader specified `Connection: close`.
+    // Technically, we shouldn't need to care if the client wants to close the connection, we could still re-use the connection from this server to leader, except that we've already sent
+    // it a command with `Connection: close` on this socket so we should expect that it will honor that and close this socket.
+    bool keepOpen = true;
+    for (const auto& message : {command.request.nameValueMap, command.response.nameValueMap}) {
+        auto connectionHeader = message.find("Connection");
+        if (connectionHeader != message.end() && connectionHeader->second == "close") {
+            keepOpen = false;
+            break;
+        }
+    }
```

Comment:
This logic is going to need to be reused in all the `runOn` methods - what do you think about putting it in a function on BedrockCommand, something like `command->shouldReuseSocket()`?

Reply:
Meh, doesn't really matter to me. Right now it's only used here. I'm fine refactoring it to there later when it's needed. Either way.